addons/mail/static/src/messaging/entity/attachment/attachment_tests.js

```diff
@@ -0,0 +1,154 @@
+odoo.define('mail.messaging.entity.AttachmentTests', function (require) {
+'use strict';
+
+const {
+    afterEach: utilsAfterEach,
+    beforeEach: utilsBeforeEach,
+    pause,
+    start: utilsStart,
+} = require('mail.messaging.testUtils');
+
+QUnit.module('mail', {}, function () {
+QUnit.module('messaging', {}, function () {
+QUnit.module('entity', {}, function () {
+QUnit.module('Attachment', {
+    beforeEach() {
+        utilsBeforeEach(this);
+        this.start = async params => {
+            if (this.widget) {
+                this.widget.destroy();
+            }
+            let { env, widget } = await utilsStart(Object.assign({}, params, {
+                data: this.data,
+            }));
+            this.env = env;
+            this.widget = widget;
+        };
+    },
+    afterEach() {
+        utilsAfterEach(this);
+        this.env = undefined;
```

Comment:
This doesn't seem necessary.

Reply:
This is to enforce a crash in case a test still makes use of `env` after teardown.
Same for `widget` below.